net/netcheck/netcheck.go

```diff
@@ -1126,7 +1126,9 @@ func (c *Client) checkCaptivePortal(ctx context.Context, dm *tailcfg.DERPMap, pr
 		return false, err
 	}
 
-	chal := "tailscale " + node.HostName
+	// Note: the set of valid characters in a challenge and the total
+	// length is limited; see the derper code for more details
```

Comment:
Maybe explicitly mention `isChallengeChar` as where this is enforced?

Reply:
Good call; done!